content/docs/thinking-in-react.md

```diff
@@ -31,118 +31,118 @@ Our JSON API returns some data that looks like this:
+Untuk membuat versi statis dari aplikasi Anda yang me-render model data, Anda akan membangun komponen yang menggunakan kembali komponen lain dan mengoper data menggunakan *props*. *props* adalah sebuah cara untuk mengoper data dari komponen induk ke komponen anak. Jika anda telah familiar dengan konsep *state*, **jangan menggunakan state sama sekali** untuk membangun versi statis ini. State disediakan hanya untuk interaktifitas, yang berarti, data yang berubah seiring waktu. Karena ini adalah versi statis dari aplikasi yang Anda buat, Anda belum membutuhkannya.
 
-You can build top-down or bottom-up. That is, you can either start with building the components higher up in the hierarchy (i.e. starting with `FilterableProductTable`) or with the ones lower in it (`ProductRow`). In simpler examples, it's usually easier to go top-down, and on larger projects, it's easier to go bottom-up and write tests as you build.
+Anda dapat membangun dari atas ke bawah atau dari bawah ke atas. Maksudnya, anda dapat mulai membangun komponen teratas dari hierarki (mis. memulai dari `FilterableProductTable`) atau dari komponen terbawah (`ProductRow`). Dalam contoh simpel, biasanya lebih mudah membangun dari atas ke bawah, dan dalam proyek yang lebih besar, biasanya lebih mudah membangun dari bawah ke atas dan menambahkan tes dalam prosesnya.
 
-At the end of this step, you'll have a library of reusable components that render your data model. The components will only have `render()` methods since this is a static version of your app. The component at the top of the hierarchy (`FilterableProductTable`) will take your data model as a prop. If you make a change to your underlying data model and call `ReactDOM.render()` again, the UI will be updated. It's easy to see how your UI is updated and where to make changes since there's nothing complicated going on. React's **one-way data flow** (also called *one-way binding*) keeps everything modular and fast.
+Pada akhir langkah ini, Anda akan memiliki sebuah library dari komponen pakai ulang yang akan me-render model data Anda. Komponen-komponen tersebut hanya akan memiliki method `render()` karena mereka adalah versis statis dari aplikasi Anda. Komponen di atas hierarki (`FilterableProductTable`) akan mengambil model data sebagai sebuah prop. Jika Anda membuat perubahan pada model data dan memanggil kembali `ReactDOM.render()`, UI akan di-update secara otomatis. Akan mudah untuk melihat bagaimana UI Anda di-update dan dimana untuk melakukan perubahan karena tidak ada hal rumit yang terjadi. Konsep **one-way data flow** (disebut juga *one-way binding*) dari React membuat segalanya modular dan cepat.
```

Comment:
Not sure if I should translate the more technical terms like "one-way data flow", "don't repeat yourself", etc. So I left them as-is.

Reply:
kalau gini tepat ngga mas?
`hanya akan memiliki method` -> `hanya memiliki method`
`UI akan di-update ` -> `UI akan diperbaharui`
`membuat segalanya modular dan cepat` -> `membuat segalanya dengan modular dan cepat`

Reply:
library -> *library*

For engineering terms like DRY & SOC I personally prefer to keep them in English for now mas. Because I think it doesn't make sense yet to be translated, but please put them in the [glossary](https://github.com/reactjs/id.reactjs.org/wiki/Glosarium), later when we're going to translate it or wants to build a more robust glossary we can easily build it 😄